src/bp_sim.cpp

```diff
@@ -3808,10 +3808,8 @@ void CNodeGenerator::handle_pcap_pkt(CGenNode *node, CFlowGenListPerThread *thre
 
 void CNodeGenerator::handle_timesync_msg(CGenNodeTimesync *node, CFlowGenListPerThread *thread, bool &exit_scheduler) {
     m_p_queue.pop();
-    if ((m_p_queue.size() == m_parent->m_non_active_nodes) ||  // exit in case this is the last node
```

Comment:
What is wrong with this condition?

Reply:
OR
It will exit scheduler in case it is last node (probably?) OR in case it is marked for free.
Both situation can but didn't meet to be meet together.
In today's test we found that timesync node is marked for free and exit scheduler and didn't free left nodes.
I just removed this line as pcap node didn't check if it is last node (I think in case this is last node Trex will handle it)

Reply:
I am not sure if I understand.